src/OAuthClient/HoldRequestClient.php

```diff
@@ -14,16 +18,16 @@ class HoldRequestClient extends APIClient
      * @return bool
      * @throws APIException
      */
-    public static function validateRequestId($holdRequestId)
+    public static function validateRequestId(int $holdRequestId)
     {
-        if (!isset($holdRequestId) || $holdRequestId === '') {
-            throw new APIException(
-                'No Hold Request Id.',
-                'No Hold Request Id provided.',
-                0,
+        if (!isset($holdRequestId) || !is_numeric($holdRequestId) || $holdRequestId < 1) {
+            throw new NonRetryableException(
+                'Not Acceptable: Invalid hold request id: ' . $holdRequestId,
+                'Not Acceptable: Invalid hold request id: ' . $holdRequestId,
+                406,
```

Comment:
Same as above for exception params; second param can be null instead of doubling the message.